src/blueteam/Robot.java

```diff
@@ -131,4 +138,29 @@ boolean willCollideWithMe(BulletInfo bullet) {
 
 		return (perpendicularDist <= rc.getType().bodyRadius);
 	}
+
+	/**
+	 * Try to move in direction perpendicular to bullet trajectory
+	 *
+	 * @param bullet
+	 * @return true if move succeeded
+	 */
+	boolean trySidestep(BulletInfo bullet) {
+		Direction towards = bullet.getDir();
+		return (tryMove(towards.rotateRightDegrees(90)) || tryMove(towards.rotateLeftDegrees(90)));
+	}
+
+	/**
+	 * Try to dodge all bullets in range.
+	 */
+	void dodge() {
+		BulletInfo[] bullets = rc.senseNearbyBullets();
+		Arrays.stream(bullets).filter(x -> willCollideWithMe(x)).forEach(x -> trySidestep(x));
```

Comment:
please refer to http://archive.battlecodeforum.org/2016/t/problem-using-java-streams/162